packages/server-wallet/src/__test__/sync-objectives.test.ts

```diff
@@ -68,6 +68,82 @@ test('Objectives can be synced if a message is lost', async () => {
   });
 });
 
+test('handles the objective being synced even if no message is lost', async () => {
+  const createChannelParams: CreateChannelParams = createChannelArgs();
+
+  // We mimic not receiving a message containing objectives
+  const messageResponse = await a.createChannel(createChannelParams);
+
+  const channelId = messageResponse.channelResults[0].channelId;
+  const objectiveId = `OpenChannel-${channelId}`;
+  await b.pushMessage(getPayloadFor(bob().participantId, messageResponse.outbox));
+
+  // We expect both objectives to be there
+  expect(await getObjective(a.knex, objectiveId)).toBeDefined();
+  expect(await getObjective(b.knex, objectiveId)).toBeDefined();
+
+  const {outbox, newObjectives, channelResults} = await a.syncObjectives([objectiveId]);
+
+  // The only result will be a message containing the latest states
+  expect(outbox).toHaveLength(1);
+  expect(outbox[0]).toMatchObject({
+    method: 'MessageQueued',
+    params: {
+      recipient: 'bob',
+      sender: 'alice',
+      data: {
+        signedStates: [expect.objectContaining({turnNum: 0})],
+        requests: [],
```

Comment:
## Real comment
Thinking through this, why should `b.pushMessage` change the behaviour of `a.syncObjective` in this case? Did somehow `a` learn that `b` has the objective in this test case?

## Meta comment
It's a bit sad that to understand the point of this test, I need to understand that
1. `requests: []` is the important thing to focus on
2. `expect(['foo']).toMatchObject([]);` fails

Reply:
Resolved in e6a1605